spec/javascripts/components/intervention-spec.js

```diff
@@ -0,0 +1,91 @@
+/* eslint-env jasmine, jquery */
+/* global GOVUK */
+
+describe('Intervention banner component', function () {
+  'use strict'
+
+  var container
+
+  beforeEach(function () {
+    container = document.createElement('div')
+    container.innerHTML =
+      '<section class="gem-c-intervention" data-module="intervention" data-intervention-name="test-intervention-name"><a class="govuk-link gem-c-intervention__dismiss-link">Dismiss</a></section>'
+
+    document.body.appendChild(container)
+    var element = document.querySelector('[data-module="intervention"]')
+    new GOVUK.Modules.Intervention(element).init()
+
+    window.GOVUK.setCookie('cookies_policy', '{"essential":true,"settings":true,"usage":true,"campaigns":true}')
+  })
+
+  afterEach(function () {
+    GOVUK.deleteCookie('intervention_campaign')
+    document.body.removeChild(container)
+  })
+
+  describe('close banner', function () {
+    it('should hide intervention banner', function () {
+      var banner = document.querySelector('.gem-c-intervention')
+      var close = document.querySelector('.gem-c-intervention__dismiss-link')
+      expect(banner).toBeVisible()
+
+      close.click()
+
+      expect(banner).toBeHidden()
+    })
+
+    it('sets a cookie value', function () {
+      spyOn(GOVUK, 'cookie').and.callThrough()
+      var close = document.querySelector('.gem-c-intervention__dismiss-link')
+      close.click()
+
+      expect(GOVUK.cookie).toHaveBeenCalled()
+      var bannerCookie = GOVUK.cookie('intervention_campaign')
+      expect(bannerCookie).toEqual('test-intervention-name')
+    })
+  })
+
+  describe('cookies are already set', function () {
+    it('does not display the banner', function () {
+      GOVUK.setCookie('intervention_campaign', 'test-intervention-name', { days: 1 })
+      var element = document.querySelector('[data-module="intervention"]')
+      new GOVUK.Modules.Intervention(element).init()
+
+      var banner = document.querySelector('.gem-c-intervention')
+
+      expect(banner).toBeHidden()
+    })
+  })
+})
+
+describe('another concurrent banner campaign', function () {
```

Comment:
Why is this an entirely separate describe block? It should be part of the main `Intervention banner component` describe block (much of the `before` and `after` code looks duplicated).

Reply:
I had it under the same block but for clarity wanted it separated. The duplication is ok in this in my view - it's only one duplication. The other thing was about making sure it was for _two_ different banners and to initialise a different use of the component.

Reply:
The main reason for keeping all tests in a single describe block is so that they're kept together when run in the browser, currently they're separate and there's no obvious indication what the other describe block is testing without grepping through the code:

![Screenshot 2023-02-20 at 14 37 40](https://user-images.githubusercontent.com/861310/220135795-780b72fa-2a81-4970-bd77-b584c827cfb5.png)

The other things you mention should be resolvable - I'd be happy to pair if that would help?